src/main/dialogs/basic/dialog.js

```diff
@@ -0,0 +1,433 @@
+ *
+ *   For example, a relative path such as `"/src/rendered/waitDialog/index.html"` might be split into:
+ *   `prefixPath = "src/renderer"`
+ *   `fileSuffixPath = "waitDialog/index.html"`
+ */
+fluid.defaults("gpii.app.dialog", {
+    gradeNames: ["fluid.modelComponent", "gpii.app.resizable"],
+
+    model: {
+        isShown: false,
+        // the positions of the window,
+        // represented as offset from the bottom right corner;
+        // by default the window is positioned in the bottom right corner
+        offset: {
+            x: 0,
+            y: 0
+        }
+    },
+
+    events: {
+        onDialogShown: null,
+        onDialogHidden: null
+    },
+
+    config: {
+        // dialog behaviour settings
+        showInactive: false,
+
+        // Whether to position the dialog after creation. This is used instead of x and y as raw
+        // options as they depend on the offset and the latter is not yet defined at the time of the
+        // dialog creation
+        positionOnInit: true,
+
+        // Whether the window is hidden offscreen and should be treated as such. Its usage is
+        // mainly assotiated with the `gpii.app.dialog.offScreenHidable` grade
+        hideOffScreen: false,
+
+        // Whether the window can be closed (which will destroy it). In case
+        // this setting is active the only way for a window to be closed is through
+        // the usage of the `destroy` method and a close command would simply hide the window.
+        // This is mainly needed to avoid closing a window using the Alf + F4 combination
+        closable: false,
+
+        restrictions: {
+            minHeight: null
+        },
+
+        // params for the BrowserWindow instance
+        params: null,
+
+        // dialog creation options
+        attrs: {        // raw attributes used in `BrowserWindow` generation
+            width: 800,
+            height: 600,
+            show: false,
+            frame: false,
+            transparent: true,
+            alwaysOnTop: true,
+            skipTaskbar: true,
+            type: "toolbar",
+            resizable: false
+        },
+        filePrefixPath: "src/renderer",
+        fileSuffixPath: null,           // e.g. "waitDialog/index.html"
+        url: {
+            expander: {
+                funcName: "gpii.app.dialog.buildFileUrl",
+                args: [
+                    "{that}.options.config.filePrefixPath",
+                    "{that}.options.config.fileSuffixPath"
+                ]
+            }
+        }
+    },
+    members: {
+        // XXX move to the model probably
+        width:  "{that}.options.config.attrs.width", // the actual width of the content
+        height: "{that}.options.config.attrs.height", // the actual height of the content
+
+        // Blurrable dialogs will have the `gradeNames` property which will contain
+        // all gradeNames of the current component. Useful when performing checks about
```

Comment:
Slightly clearer changing current -> containing. Also, make explicit reference to BrowserWindow.getFocusedWindow and the file blurrable.js to help the reader.